src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/StreamExtensions.cs

```diff
@@ -0,0 +1,33 @@
+#nullable enable
+using System;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace GovUk.Education.ExploreEducationStatistics.Common.Extensions
+{
+    public static class StreamExtensions
+    {
+        public static void WriteText(this Stream stream, string text)
+        {
+            using var writer = new StreamWriter(stream, leaveOpen: true);
+            writer.Write(text);
+            writer.Flush();
+        }
+
+        public static string ReadToEnd(this Stream stream)
+        {
+            using var reader = new StreamReader(stream, leaveOpen: true);
+            return reader.ReadToEnd();
+        }
+
+        public static string ComputeMd5Hash(this Stream stream)
```

Comment:
Just wondering where we use this? I think I've only seen creating the unique zip filename by String on this PR, not by stream.

Reply:
Ah good catch. I actually forgot to use this in the HTTP caching - it's now being used in `CacheWithETag` to prevent user's re-requesting the same file over and over (at least in the browser).

Reply:
Cool, just seen this update.